prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -209,6 +223,37 @@ def __call__(self, summary: str) -> str:
         return summary
 
 
+class GetWalletBalance(Function):
+    @property
+    def description(self) -> str:
+        return "Use this function to fetch the balance of a user, given in xDAI units."
+
+    @property
+    def example_args(self) -> list[str]:
+        return ["0x2DD9f5678484C1F59F97eD334725858b938B4102"]
```

Comment:
I don't the agent needs to be aware of its own wallet address, so I think this function can take no arguments, and just internally use the `PUBLIC_KEY` global variable.